Core/MigrationOptions.cs

```
@@ -30,5 +35,40 @@ public class MigrationOptions
     public Func<Document, bool> SourceFileFilter { get; } = _ => true;
 
     public Encoding Encoding { get; } = Encoding.Default;
+
+    public IEnumerable<IDocumentConverter> Converters { get; } = new List<IDocumentConverter>();
+
+    private MigrationOptions ()
+    {
+    }
+
+    private MigrationOptions (
+        Func<Project, bool> projectFilter,
+        Func<Document, bool> documentFilter,
+        Encoding encoding,
+        IEnumerable<IDocumentConverter> converters)
+    {
+      ProjectFilter = projectFilter;
+      SourceFileFilter = documentFilter;
+      Encoding = encoding;
+      Converters = converters;
+    }
+
+    public MigrationOptions WithConverters (IEnumerable<string> converterNames)
+    {
+      var converter = converterNames.Select (CreateConverter).WhereNotNull();
+
+      return new MigrationOptions (ProjectFilter, SourceFileFilter, Encoding, converter);
+    }
+
+    private IDocumentConverter? CreateConverter (string conversionFeatureName)
+    {
+      return conversionFeatureName switch
+      {
+          "expected-exception" => new ExpectedExceptionDocumentConverter(),
+          "asserts" => new AssertRenamingDocumentConverter(),
+          _ => null,

```

I guess this should throw an exception instead of silently ignoring our argument in case of a typo.